src/openforms/logging/logevent.py

```diff
@@ -439,20 +447,97 @@ def submission_export_list(form: "Form", user: "User"):
 
 
 def submission_logic_evaluated(
-    submission: "Submission", evaluated_rules, updated_submission_data
+    submission: "Submission", evaluated_rules, resulting_data
 ):
-
     if not evaluated_rules:
         return
     log_evaluated_rules = []
-
+    formsteps = submission.form.formstep_set.select_related("form_definition").all()
+    # Mapping each form's key to :
+    # - the form_step it belongs
+    # - The component related to it
+    keys_mapped_to_form = {}
+    keys_mapped_to_component = {}
+    for formstep in formsteps:
+        keys_mapped_to_form.update(
+            {
+                component["key"]: formstep
+                for component in formstep.form_definition.iter_components()
+            }
+        )
+        keys_mapped_to_component.update(flatten(formstep.form_definition.configuration))
+
+    involved_components = {}
     for evaluated_rule in evaluated_rules:
 
         rule = evaluated_rule["rule"]
+        logic_data = JsonLogicTest.from_expression(rule.json_logic_trigger)
+        source_components = convert_json_logic_in_json_tree(
+            logic_data,
+            keys_mapped_to_component,
+            keys_mapped_to_form,
+            resulting_data,
+        )
+
+        stringified_json_tree = stringify_json_tree(source_components)
+        print(stringified_json_tree)
+
+        # will remove duplicate components that are involved in several rules
+        remove_duplicate_components_from_json_tree(
+            source_components, involved_components
+        )
+        targeted_components = []
+        for action in rule.actions:
```

Comment:
In this loop, you are doing a lot of "extra" work that is probably not needed. During the loop, you basically set up a bunch of data and add it to `targeted_components`, and then you post-process `targeted_components` before passing it to the evaluated rule data.

It'd probably make more sense to do all the processing during the for loop already and convert it in exactly the data structure you want, e.g.:

```py
        all_variables = submission.form.formvariable_set.in_bulk(field_name="key")  # move this OUTSIDE of the parent loop - it only needs to be calculated once.
        targeted_components = []
        component_key_lookup = {
            LogicActionTypes.value: "component",
            LogicActionTypes.property: "component",
            LogicActionTypes.variable: "variable",
        }
        for action in rule.actions:
            action_details = action["action"]
            component_key = component_key_lookup.get(action_details["type"])
            component_meta = components_map.get(component_key)
            component = component_meta[1] if component_meta else None

            # figure out the best possible label
            # 1. fall back to component label if there is a label, else empty string
            # 2. if there is a variable, use the name if it's set, else fall back to 
            #    component label
            label = component.get("label", "") if component else "" 
            if component_key in all_variables:
                label = all_variables[component_key].name or label
            
            action_log_data = {
                "type": action_details["type"],
                "type_display": LogicActionTypes.labels[action_details["type"]],
                "label": label or "",
                "step_name": component_meta[0].form_definition.name if component_meta else "",
                "value": "",
            }

            # process the value
            if action_details["type"] == LogicActionTypes.value:
                action_log_data["value"] = action_details["value"]
            elif action_details["type"] == LogicActionTypes.property:
                action_log_data.update({
                    "value": action_details["property"]["value"],
                    "state": action_details["state"],
                })
            elif action_details["type"] == LogicActionTypes.variable:
                action_logic_introspection = introspect_json_logic(action_details["value"], components_map, resulting_data)
                action_log_data["value"] = action_logic_introspection.as_string()
            target_components.append(action_log_data)
```

In this form, it should be good to directly include in the log record extra data.

Reply:
It seems that we can't use in_bulk() on not unique fields, and key isn't one